pkg/agent/cniserver/ipam/antrea_ipam_test.go

```diff
@@ -532,3 +532,55 @@ func TestAntreaIPAMDriver(t *testing.T) {
 	// Make sure repeated call without previous container results in error
 	testAddError("pear3")
 }
+
+func TestSecondaryNetworkAdd(t *testing.T) {
+	testCases := []struct {
+		name        string
+		networkConf *argtypes.NetworkConfig
+		args        *invoke.Args
+		k8sArgs     *argtypes.K8sArgs
+		expectedRes error
+	}{
+		{
+			name: "Both Antrea IPPool and static address is empty",
+			networkConf: &argtypes.NetworkConfig{
+				CNIVersion: "0.4.0",
+				IPAM: &argtypes.IPAMConfig{
+					IPPools: []string{},
+				},
+			},
+			expectedRes: fmt.Errorf("at least one Antrea IPPool or static address must be specified"),
+		},
+		{
+			name: "Specify the static IP Address",
+			networkConf: &argtypes.NetworkConfig{
+				CNIVersion: "0.4.0",
+				IPAM: &argtypes.IPAMConfig{
+					Addresses: []argtypes.Address{
+						{Address: "192.168.1.1/24"},
+					},
+				},
+			},
+		},
+		{
+			name: "Antrea IPAM driver not ready",
+			networkConf: &argtypes.NetworkConfig{
+				CNIVersion: "0.4.0",
+				IPAM: &argtypes.IPAMConfig{
+					IPPools: []string{
+						"192.168.1.1/16",
+					},
+				},
+			},
+			expectedRes: fmt.Errorf("Antrea IPAM driver not ready: timed out waiting for the condition"),
+		},
+	}
+
+	for _, tt := range testCases {
+		t.Run(tt.name, func(t *testing.T) {
+			d := &AntreaIPAM{}
+			_, err := d.secondaryNetworkAdd(tt.args, tt.k8sArgs, tt.networkConf)
+			assert.Equal(t, tt.expectedRes, err)
```

Comment:
Seems the test only cover the error cases, is it possble to add the logic for happy path, so that we can have a test to ensure the complete work flow is able to walk through?

Reply:
Added successful case.